src/modules/module-switch-on-connect.c

```diff
@@ -70,11 +70,14 @@ static pa_hook_result_t sink_put_hook_callback(pa_core *c, pa_sink *sink, void*
 
     pa_log_debug("Trying to switch to new sink %s", sink->name);
 
-    /* Don't switch to any internal devices */
-    s = pa_proplist_gets(sink->proplist, PA_PROP_DEVICE_BUS);
-    if (pa_safe_streq(s, "pci") || pa_safe_streq(s, "isa")) {
-        pa_log_debug("Refusing to switch to sink on %s bus", s);
-        return PA_HOOK_OK;
+    /* Don't switch to any internal devices except HDMI */
```

Comment:
Can you add a commit message here? I don't understand why some devices would be excluded from the switching (i.e. I don't even understand the original code)

Reply:
Makes sense. I'm surprised that the code has survived so far in that state (regarding HDMI as an internal device).